### DISCUSSION TITLE
Normal that real time for training increases ?

### DISCUSSION THREAD

**Discussion by Matcom274** [mir-group/nequip#468] [Q&A] [answered]
Literally, is it normal that the training time per epoch (real time or wall time) increases as training evolves just like the attached image. 

I have tested with 5 models with adjusting several parameter values such as l_max, rcutoff, even though these parameters migh not be  related... The drop of the line indicates when I restart the training using append=true setting. 
It seems the usage of gpu memory slightly increased after 20 epoch but saturated at last. Would it be the reason? I thought that the model complexity is initially fixed (fixed memory) and the weights are the only trainable values. Maybe I don't fully understand how GNN exactly works.

![tr_time-epoch](https://github.com/user-attachments/assets/84e70335-37a0-4819-a05a-bd28a33b1a41)
* wall time and epoch from 'metrics_epoch.csv' log file.

settings
- almost the same with example.yaml 
- training data is my own around ~5000frames with 70 atoms per cell.
- cuda11.3 / torch1.11.0 / A6000 or rtx3090

Thank you for your help in advance..! 
Sorry if this question has been already discussed. I tried to search but couldn't find it.

**Comment by Matcom274** [comment, marked as answer]
Oh.. I just noticed it is accumulated time.

**Comment by Linux-cpp-lisp** [comment]
Hi @Matcom274 ,

Correct, it is cumulative time; this looks normal. 

That said, there has previously been reported https://github.com/mir-group/nequip/discussions/311, so if you ever observe _that_ issue, please do post in that discussion.
